packages/runtime/datastore/src/channelStorageService.ts

```diff
@@ -39,13 +40,19 @@ export class ChannelStorageService implements IChannelStorageService {
         return this.flattenedTree[path] !== undefined;
     }
 
-    public async read(path: string): Promise<string> {
+    public async readBlob(path: string): Promise<IBlob> {
         const id = await this.getIdForPath(path);
-        const blob = this.extraBlobs !== undefined
-            ? this.extraBlobs.get(id)
-            : undefined;
+        const blob = this.extraBlobs?.get(id) ?? this.storage.readBlob(id);
+
+        return blob;
+    }
 
-        return blob ?? this.storage.read(id);
+    public async read(path: string): Promise<string> {
```

Comment:
It is safe to remove read() on this object completely, as it is in the same layer as its consumers, i.e. there is no backward compatibility concerns. As long as we expose readBlob() and easy way to consume strings (via readString() API no this class, or a stand-alone helper function), we should be good.
Removing read() here (and especially from interfaces) will allow compiler to help us see if we missed any cases, and thus will decrease chances we will mix incorrectly different types (decoded and non-decoded strings). I.e. if there are no read() and no fromBase64ToUtf8() calls in runtime code, then most likely we are doing conversions correctly